### ISSUE TITLE
Can't take unmanaged generic class as function argument

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#9580]
### Summary of Problem
We encounter an internal compiler error when a function declares that its argument is unmanaged but also the generic version of the type.  This seems like an unfortunate backslide in functionality when compared to the old strategy for classes.

### Steps to Reproduce

**Source Code:**

```chapel
class Foo {
  type t;
  var v: int;
}

proc takesUnmanagedFoo(x: unmanaged Foo) {
  writeln(x);
}

var foo1 = new unmanaged Foo(int, 3);
var foo2 = new unmanaged Foo(real, 4);
takesUnmanagedFoo(foo1);
takesUnmanagedFoo(foo2);
delete foo1, foo2;
```

**Compile command:**
`chpl unmanaged-generic2.chpl`

**Associated Future Test(s):**
[`test/classes/delete-free/unmanaged/unmanaged-generic2.chpl`](https://github.com/chapel-lang/chapel/blob/master/test/classes/delete-free/unmanaged/unmanaged-generic2.chpl) #9581

### Configuration Information

- Output of `chpl --version`: chapel 1.18.0 (pre-release)

**Comment by lydia-duncan** [comment]
@mppf - heads up, I think this is causing one of the default initializer failures (and I encountered it in another setting)

**Comment by lydia-duncan** [comment]
I suspect owned and shared are similarly impacted

**Comment by mppf** [comment]
Yes this is a known issue in the backlog -- #9521

**Comment by lydia-duncan** [comment]
This one doesn't involve a query, but the examples I see in that issue all do, so I think this is a separate (but related) issue

**Comment by lydia-duncan** [comment]
I'll close the duplicate of that one I just opened, sorry

**Comment by mppf** [comment]
Yeah, anyway thanks for the .futures. I intend to fix both at once when I work on #9521.

**Comment by lydia-duncan** [comment]
Sounds good!  Would you mind checking to make sure they are otherwise written correctly, since I'm not used to writing for owned/shared/unmanaged?
